spec/lib/ruby_smb/dcerpc/response_spec.rb

```diff
@@ -131,7 +131,7 @@
   end
 
   describe '#stub_length' do
-    let(:stub_length) { rand(0xFF) }
+    let(:stub_length) { rand(1..0xFF) }
```

Comment:
It looks like stub length can be 0:

> For PDUs that contain stub data, the header is padded, if necessary, with zeros to an integral multiple of 8 octets in length. This allows the stub data to assume it will always begin with 0 MOD 8 alignment. Padding, where necessary, is included in the type declarations.
> https://pubs.opengroup.org/onlinepubs/9629399/chap12.htm#Alignment

Is this an actual bug in the implementation?

Reply:
Yes, absolutely. It seems to only fail when `:auth_size` is zero. Setting `:stub_size` to zero works in my environment (`let(:stub_size) { 0 }`).

The reason it fails are probably because authentication verifier are set even if `:auth_size` is 0, which makes something failing. We might want to had some checks for this corner case. In the meantime, I rewrote these specs and make sure the zero-size stub is taken in consideration:

https://github.com/cdelafuente-r7/ruby_smb/blob/41232dd39feb863bb57aceb4260b1ed488556761/spec/lib/ruby_smb/dcerpc/response_spec.rb#L133-L221

Please, let me know if this makes sense and I will put a PR for it.

Reply:
Awesome, thanks for taking a look! A pull request sounds good for that 💯